sdk/core/core/test/cmocka/test_az_http.c

```diff
@@ -512,6 +521,33 @@ static void test_http_response_header_validation_space(void** state)
   }
 }
 
+static void test_http_response_append(void** state)
+{
+  (void)state;
+  {
+    uint8_t buffer[10];
+    az_http_response response = { 0 };
+    az_span append_this = AZ_SPAN_FROM_STR("0123456789");
+    assert_return_code(az_http_response_init(&response, AZ_SPAN_FROM_BUFFER(buffer)), AZ_OK);
+    assert_return_code(az_http_response_append(&response, append_this), AZ_OK);
+    assert_memory_equal(buffer, az_span_ptr(append_this), 10);
+  }
+}
+
+static void test_http_response_append_overflow(void** state)
+{
+  (void)state;
+  {
+    uint8_t buffer[10];
+    az_http_response response = { 0 };
+    az_span append_this = AZ_SPAN_FROM_STR("0123456789123456");
+
+    assert_return_code(az_http_response_init(&response, AZ_SPAN_FROM_BUFFER(buffer)), AZ_OK);
+    az_result overflow = az_http_response_append(&response, append_this);
```

Comment:
nit: Don't call this overflow, since that assumes the result.
```suggestion
    az_result result = az_http_response_append(&response, append_this);
```

Reply:
Also nit: Check that the buffer is still empty after this fails.